bin/base.sh

```diff
@@ -27,6 +27,9 @@ builder_choose () {
 }
 
 parse_basic_args () {
+    if [ "$#" -eq 0 ]; then
+        usage 0
```

Comment:
I believe standard practice is to exit with failure if there is a usage error, which no arguments is, even if we are also printing help.